src/main/java/teamrtg/rtg/core/event/EventManagerRTG.java

```diff
@@ -33,238 +35,349 @@
     }
+    
+    public class InitBiomeGensRTG
+    {
+        InitBiomeGensRTG()
+        {
+            Logger.debug(EVENT_SYSTEM + "Initialising InitBiomeGensRTG");
+        }
 
-    @SubscribeEvent
-    public void onBiomeGenInit(WorldTypeEvent.InitBiomeGens event) {
-
-        // only handle RTG world type
-        if (!event.getWorldType().getWorldTypeName().equalsIgnoreCase("RTG")) return;
+        @SubscribeEvent
+        public void initBiomeGensRTG(WorldTypeEvent.InitBiomeGens event) {
 
-        boolean stripRivers = true; // This used to be a config option. Hardcoding until we have a need for the option.
+            // only handle RTG world type
+            if (!event.getWorldType().getWorldTypeName().equalsIgnoreCase("RTG")) return;
 
-        if (stripRivers) {
             try {
                 event.setNewBiomeGens(new RiverRemover().riverLess(event.getOriginalBiomeGens()));
             } catch (ClassCastException ex) {
-                //throw ex;
                 // failed attempt because the GenLayers don't end with GenLayerRiverMix
             }
         }
     }
+    
+    public class InitMapGenRTG
+    {
+        InitMapGenRTG() {
+            Logger.debug(EVENT_SYSTEM + "Initialising InitMapGenRTG");
+        }
+
+        @SubscribeEvent(priority = EventPriority.LOW)
+        public void initMapGenRTG(InitMapGenEvent event) {
+
+            Logger.debug("event type = %s", event.getType().toString());
+            Logger.debug("event originalGen = %s", event.getOriginalGen().toString());
 
-    @SubscribeEvent
-    public void onWorldLoad(WorldEvent.Load event) {
-        if ((event.getWorld().getWorldInfo().getTerrainType() instanceof WorldTypeRTG)) {
-            if (event.getWorld().provider.getDimension() == 0) {
-                Logger.info("World Seed: %d", event.getWorld().getSeed());
-                Logger.info("Reloading configs");
-                Mods.syncAllConfigs();
+            if (event.getType() == InitMapGenEvent.EventType.SCATTERED_FEATURE) {
+                event.setNewGen(new MapGenScatteredFeatureRTG());
+            } else if (event.getType() == InitMapGenEvent.EventType.VILLAGE) {
+
+                if (Mods.RTG.config.ENABLE_VILLAGE_MODIFICATIONS.get()) {
+                    event.setNewGen(new MapGenVillageRTG());
+                }
+            } else if (event.getType() == InitMapGenEvent.EventType.CAVE) {
+
+                if (Mods.RTG.config.ENABLE_CAVE_MODIFICATIONS.get()) {
+
+                    event.setNewGen(new MapGenCavesRTG());
+                }
+            } else if (event.getType() == InitMapGenEvent.EventType.RAVINE) {
+
+                if (Mods.RTG.config.ENABLE_RAVINE_MODIFICATIONS.get()) {
+
+                    event.setNewGen(new MapGenRavineRTG());
+                }
+            } else if (event.getType() == InitMapGenEvent.EventType.OCEAN_MONUMENT) {
+                event.setNewGen(new StructureOceanMonumentRTG());
             }
```

Comment:
This would probably be better organised and more efficient as a switch statement as I did in 1.7.10: [EventManagerRTG.java#L148-L173](https://github.com/Team-RTG/Realistic-Terrain-Generation/blob/cbc148503775c972b6599e78b95b9a18af8999db/src/main/java/rtg/event/EventManagerRTG.java#L148-L173)
Note: The missing cases can also be added and left blank for future reference if we ever need to do anything with other InitMapGenEvent events.